packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -2686,7 +2686,198 @@ describe('NetworkController', () => {
           );
         });
       });
+      describe('if the controller was destroyed and the all listeners were removed', () => {
+        it('lookupNetwork does not thrown an error', async () => {
+          await withController(
+            {
+              state: {
+                providerConfig: buildProviderConfig({
+                  type: NetworkType.rpc,
+                  chainId: toHex(1337),
+                  rpcUrl: 'https://mock-rpc-url',
+                }),
+              },
+              infuraProjectId: 'some-infura-project-id',
+            },
+            async ({ controller, messenger }) => {
+              const fakeProviders = [
+                buildFakeProvider([
+                  {
+                    request: {
+                      method: 'eth_getBlockByNumber',
+                    },
+                    response: SUCCESSFUL_ETH_GET_BLOCK_BY_NUMBER_RESPONSE,
+                    beforeCompleting: async () => {
+                      messenger.clearSubscriptions();
+                      await controller.destroy();
+                    },
+                  },
+                ]),
+              ];
+              const fakeNetworkClients = [buildFakeClient(fakeProviders[0])];
+              mockCreateNetworkClient()
+                .calledWith({
+                  chainId: toHex(1337),
+                  rpcUrl: 'https://mock-rpc-url',
+                  type: NetworkClientType.Custom,
+                  ticker: 'TEST',
+                })
+                .mockReturnValue(fakeNetworkClients[0]);
+
+              expect(controller.initializeProvider()).toBeFulfilled();
+            },
+          );
+        });
+
+        it('lookupNetwork keeps the state intact', async () => {
```

Comment:
Hmm, perhaps saying that it keeps the state intact is inaccurate, since it does change the state. I might phrase it like this:

``` suggestion
        it('does not fetch the status of the network', async () => {